pilot/pkg/controller/workloadentry/workloadentry_controller.go

```diff
@@ -619,8 +620,8 @@ func workloadEntryFromGroup(name string, proxy *model.Proxy, groupCfg *config.Co
 	if proxy.Metadata.Network != "" {
 		entry.Network = proxy.Metadata.Network
 	}
-	if proxy.Locality != nil {
-		entry.Locality = util.LocalityToString(proxy.Locality)
+	if node.Locality != nil {
+		entry.Locality = util.LocalityToString(node.Locality)
```

Comment:
This is right in case in pilot-agent, it is set correctly. I am not sure it does.

Reply:
You mean the node's locality may not always be set? In that case, the locality of the WorkloadGroup template (if any) would be used, right?

Reply:
The workload group in all other cases, configures the proxy directly I think? And then is passed to istiod via XDS.

So this seems different than that pattern, by dynamically reading it from the WG. Should we instead make the agent get the locality from the WG in the istioctl workgroup command?

Reply:
We're not currently using the istioctl workload commands. Our use case is 1 WorkloadGroup that acts as a template for WorkloadEntries spanning multiple localities, so we need to override it dynamically. I'd argue that it is currently already possible to override fields from WorkloadGroup template dynamically, e.g. labels, annotations, network and readinessprobe.

Reply:
Is it possible for users to have no WLG?

What we want is an end-to-end solution, this is not complete. But maybe add a followup?

Reply:
Hmm the code paths changed in this PR are just for autoregistration, so a WorkloadGroup is expected: https://github.com/istio/istio/blob/67362755b7ca63ff1442d627a733627ed5c75df4/pilot/pkg/controller/workloadentry/workloadentry_controller.go#L260-L265

Happy to discuss any potential followups that should be added.